src/main/java/io/kuzzle/sdk/core/KuzzleDataCollection.java

```diff
@@ -61,6 +61,17 @@ public KuzzleDataCollection(@NonNull final Kuzzle kuzzle, @NonNull final String
     }
   }
 
+  @Deprecated
+  public void advancedSearch(final JSONObject filter, final KuzzleResponseListener<KuzzleDocumentList> listener) {
+    this.advancedSearch(filter, null, listener);
+  }
+
+  @Deprecated
+  public void advancedSearch(final JSONObject filters, final KuzzleOptions options, @NonNull final KuzzleResponseListener<KuzzleDocumentList> listener) {
+    System.out.println("The advancedSearch method is now deprecated, please use the search method");
```

Comment:
This message should be printed only once, to avoid performance degradation and log spamming.